runtime/src/accounts_db.rs

```diff
@@ -3821,6 +3821,9 @@ impl AccountsDb {
     }
 
     fn do_shrink_slot_store(&self, slot: Slot, store: &Arc<AccountStorageEntry>) -> usize {
+        if self.accounts_cache.contains(slot) {
+            return 0;
+        }
```

Comment:
We cannot shrink while a slot is in the cache. When the cache slot is flushed, it adds the slot to `uncleaned_roots`. This causes clean to visit it. When clean visits a slot, it adds things to `shrink_candidate_slots`. `shrink_candidate_slots()` uses `shrink_candidate_slots` to figure out what to shrink. So, skipping a call to 'shrink' before it has been flushed is 'correct'.

Reply:
Should we assert here? We could. This would mean we would have to rewrite the at least 1 test that does this and we may find other intermittent tests. I considered rewriting the test to wait to shrink a given slot until it wasn't in the write cache. But, I didn't have control over per-slot shrinking. So, I needed to modify the api `pub fn shrink_all_slots(&self, is_startup: bool, last_full_snapshot_slot: Option<Slot>) {`. Seems like the simplest thing is to just ignore invalid calls to shrink while a slot is in the write cache.

Reply:
yep. done.